src/Parallel/ParallelComponentHelpers.hpp

```diff
@@ -82,6 +116,81 @@ using get_const_global_cache_tags =
             typename Metavariables::component_list,
             detail::get_const_global_cache_tags_from_pdal<tmpl::_1>>>>>;
 
+/*!
+ * \ingroup ParallelGroup
+ * \brief Given the metavariables, get a list of the unique tags that will
+ * specify the mutable items in the GlobalCache.
+ */
+template <typename Metavariables>
+using get_mutable_global_cache_tags =
+    tmpl::remove_duplicates<tmpl::flatten<tmpl::list<
+        typename detail::get_mutable_global_cache_tags_from_parallel_struct<
+            Metavariables>::type,
+        tmpl::transform<
+            typename Metavariables::component_list,
+            detail::get_mutable_global_cache_tags_from_parallel_struct<
+                tmpl::_1>>,
+        tmpl::transform<
+            typename Metavariables::component_list,
+            detail::get_mutable_global_cache_tags_from_pdal<tmpl::_1>>>>>;
+
+template <typename Tag>
+struct MutableCacheTag {
+  using tag  = Tag;
+  using type = std::tuple<typename Tag::type, std::vector<CkCallback>>;
+};
+
+template <typename Tag>
+struct get_mutable_cache_tag {
+  using type = MutableCacheTag<Tag>;
+};
+
+template <typename Metavariables>
+using get_mutable_global_cache_tag_storage =
+    tmpl::transform<get_mutable_global_cache_tags<Metavariables>,
+                    get_mutable_cache_tag<tmpl::_1>>;
+
+namespace GlobalCache_detail {
+
+template <typename T>
+struct type_for_get_helper {
+  using type = T;
+};
+
+template <typename T, typename D>
+struct type_for_get_helper<std::unique_ptr<T, D>> {
+  using type = T;
+};
+
+// This struct provides a better error message if
+// an unknown tag is requested from the GlobalCache.
+template <typename GlobalCacheTag, typename ListOfPossibleTags>
+struct list_of_matching_tags_helper {
+  using type = tmpl::filter<ListOfPossibleTags,
+               std::is_base_of<tmpl::pin<GlobalCacheTag>, tmpl::_1>>;
+  static_assert(not std::is_same_v<type, tmpl::list<>>,
+                "Trying to get a nonexistent tag from the GlobalCache. "
+                "To diagnose the problem, search for "
+                "'list_of_matching_tags_helper' in the error message. "
+                "The first template parameter of "
+                "'list_of_matching_tags_helper' is the requested tag, and "
+                "the second template parameter is a tmpl::list of all the "
+                "tags in the GlobalCache.  One possible bug that may "
+                "lead to this error message is a missing or misspelled "
+                "const_global_cache_tags or mutable_global_cache_tags "
+                "type alias.");
+};
```

Comment:
Just checking: you want me to change `list_of_matching_tags::type` so that it is no longer a list?.  Then presumably `list_of_matching_tags` should be renamed to `matching_tag`, and all the metafunctions that use `list_of_matching_tags` should be appropriately modified and renamed.  

I'm ok with doing this, but the reason I ask is that I did not write `list_of_matching_tags`: I just moved `list_of_matching_tags` from GlobalCache.hpp to ParallelComponentHelpers.hpp (because now GlobalCache.ci needs things that depend on `list_of_matching_tags`).

Reply:
The suggestion is:
```cpp
using list_of_matching_tags = ...;
static_assert(tmpl::size<list_of_matching_tags>::value == 1, "found more than one matching tag");
using type = tmpl::front<found_tags>;
```
i.e. a reodering of the statements, I believe. But GitHub can't find the old commit and the preview is a bit short.

In the end it does what you are suggesting, I think. The call sites can just do:
```cpp
using tag = matching_tag<Tag, TagsList>; // or whatever the syntax is
```
But we can do the check internally here rather than doing the check everywhere. Does that make sense?

Reply:
Yup, I can do that; just wanted to clarify what you were asking for, since this changes (in an easy way, they take an element not a list) a bunch of metafunctions that were already there before I added mutable stuff to the GlobalCache.